coordinator/src/main/scala/filodb.coordinator/queryengine2/QueryEngine.scala

```diff
@@ -203,10 +203,22 @@ class QueryEngine(dataset: Dataset,
      *
      * Starting off with solution 1 first until (2) or some other approach is decided on.
      */
-    toReduce.plans.foreach(_.addRangeVectorTransformer(AggregateMapReduce(lp.operator, lp.params, lp.without, lp.by)))
-    // One could do another level of aggregation per node too. Ignoring for now
-    val reduceDispatcher = pickDispatcher(toReduce.plans)
-    val reducer = ReduceAggregateExec(queryId, reduceDispatcher, toReduce.plans, lp.operator, lp.params)
+    toReduceLevel1.plans.foreach(
+      _.addRangeVectorTransformer(AggregateMapReduce(lp.operator, lp.params, lp.without, lp.by))
+    )
+
+    val toReduceLevel2 =
+      if (toReduceLevel1.plans.size >= 16) {
```

Comment:
Should it be configurable ? So that it can be dynamically adjusted ?

Reply:
+1 to make configurable

Reply:
We could make it configurable but with more changes - QueryEngine does not have server configuration right now. 

I didn't see the value in making it configurable yet. This value doesn't change from environment to environment. Also, default config change has the same process as code change.